src/java/org/apache/cassandra/index/sai/disk/PrimaryKeyMap.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import org.apache.cassandra.index.sai.SSTableQueryContext;
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+
+/**
+ * A bidirectional map of {@link PrimaryKey} to row Id.
```

Comment:
Need to document tread-safety properties.

The interface suggests this is a read-only map, so a user may accidentally conclude it is safe to use it in multithreaded context. However the implementations are not fully thread safe - there is shared, mutable, non synchronized, state involved in implementations of some methods.

It is of course ok to have a non-thread safe map, but if that was on purpose, the unsafety should be clearly visible.